test/connectivity: Add early return when deploying perf resources

learnitall opened the pull request:
This commit reorganizes ConnectivityTest.deploy and ConnectivityTest.validateDeployment to ensure that when the `--perf` flag is given from the user, only performance-related deployments are created and validated. Prior, all resources within the connectivity test are created and validated, even though they aren't used by the performance tests. This change should lead to faster startup times for performance tests.

This commit also adjusts how perf-related deployments are returned from `ConnectivityTest.deploymentList`. Prior, client deployments were returned in `srcList` and server deployments were returned in `dstList`, implying that `srcList` and `dstList` represent deployments belonging to clients and servers respectively. After review, this is an incorrect assumption that was made, as `srcList` and `dstList` split deployments for multi-cluster uses (ie source and destination cluster). Also, a deployment used during performance tests was missing from this list.

Acts as a quick way to address the issues discussed in https://github.com/cilium/cilium-cli/issues/1111 without a refactor.

Signed-off-by: Ryan Drew <ryan.drew@isovalent.com>

tklauser commented on connectivity/check/deployment.go:
```diff
@@ -766,9 +767,11 @@ func (ct *ConnectivityTest) deploymentList() (srcList []string, dstList []string
 	} else {
-		perfDeploymentNameManager := newPerfDeploymentNameManager(&ct.params)
-		srcList = []string{perfDeploymentNameManager.ClientName()}
-		dstList = append(dstList, perfDeploymentNameManager.ServerName())
```
Now that `dstList` is returned empty for performance tests, we should probably skip the `ct.waitForDeployments` call on empty `dstDeployments` here:

https://github.com/cilium/cilium-cli/blob/0b7737bce14fbc7ace66aefc9973013b19ed4e64/connectivity/check/deployment.go#L823

to avoid logging a message with an empty liste of deployments:

https://github.com/cilium/cilium-cli/blob/0b7737bce14fbc7ace66aefc9973013b19ed4e64/connectivity/check/deployment.go#L1153

tklauser reviewed (commented):
This change is rather hard to review as it moves large code blocks around but also makes some inline changes described in the commit message, i.e. the following bit:

> This commit also adjusts how perf-related deployments are returned
from `ConnectivityTest.deploymentList`. Prior, client deployments
were returned in `srcList` and server deployments were returned
in `dstList`, implying that `srcList` and `dstList` represent
deployments belonging to clients and servers respectively.
After review, this is an incorrect assumption that was made, as
`srcList` and `dstList` split deployments for multi-cluster uses
(ie source and destination cluster). Also, a deployment
used during performance tests was missing from this list.

Would it be possible to split these changes into a separate commit?

learnitall commented:
Ok did a force push to adjust the PR with your feedback @tklauser - thank you for taking a look! I split the origin commit of the PR into two and added a length check on lists returned from `ConnectivityTest.deploymentList` in `ConnectivityTest.validateDeployments`.

tklauser reviewed (approved):
LGTM, thanks Ryan!
